### DISCUSSION TITLE
Burning Crusade classic

### DISCUSSION THREAD

**Discussion by torkus** [layday/instawow#72] [Q&A]
hey! I didn't get invited to the BC-classic beta (typical) any idea how the addons are behaving this time around?

**Comment by layday** [comment]
I haven't been invited either, but I've been told TBC add-ons are all compatible with retail, and also that the beta prefix is `_classic_beta_`, which would suggest that they are repurposing `_classic_` for TBC.

**Reply by torkus** [reply]
well that would be convenient for everybody.

**Comment by layday** [comment]
Curse have added a new 'game version' for TBC add-ons. For the time being these have a `gameVersionFlavor` value of `null` and a `gameVersion` of `"2.5.1"` in the API.  An example of an add-on with TBC files is https://www.curseforge.com/wow/addons/plater-nameplates (https://addons-ecs.forgesvc.net/api/v2/addon/100547).

**Reply by layday** [reply]
It looks like you implemented the game version check in https://github.com/ogri-la/strongbox/blob/develop/src/strongbox/utils.clj#L291-L297 the same way I had, so you might be throwing TBC add-ons in with retail.

**Reply by layday** [reply]
Yep, I just tried it in strongbox and it installed the TBC version of Plater, which is the same issue I have in instawow.

**Comment by torkus** [comment]
> gameVersionFlavor value of `null`

IIRC we can't trust that field to be correct because of addons supporting both (vanilla) classic and retail. Always null is probably the best thing for it ... unless Curseforge decide to drop support for single releases supporting multiple 'flavors'.

> and a gameVersion of "2.5.1"

urgh, this is going to bite them.

Any other addons that you know of supporting classic-bc? 

I'll take a look tomorrow and see how much this will affect strongbox. My holidays started a few hours ago.

**Reply by torkus** [reply]
> so you might be throwing TBC add-ons in with retail.

yup, totally am. I'm overdue for a release, I'll see if I can fix this one tonight.

**Reply by layday** [reply]
> Any other addons that you know of supporting classic-bc?

All of these add-ons have 2.5 releases, although some of them are not categorised as belonging to TBC:

```
5168 https://www.curseforge.com/wow/addons/baud-mark
457334 https://www.curseforge.com/wow/addons/craftpresence
3358 https://www.curseforge.com/wow/addons/deadly-boss-mods
289311 https://www.curseforge.com/wow/addons/deadly-boss-mods-dbm-event-sound-pack
267646 https://www.curseforge.com/wow/addons/deadly-boss-mods-dbm-heroes-of-the-storm-hots
267605 https://www.curseforge.com/wow/addons/deadly-boss-mods-dbm-overwatch-countdown-pack
61882 https://www.curseforge.com/wow/addons/dbm-pvp
289632 https://www.curseforge.com/wow/addons/deadly-boss-mods-dbm-smoothmcgroove-event-pack
292896 https://www.curseforge.com/wow/addons/dejaauras
44708 https://www.curseforge.com/wow/addons/dejamark
338591 https://www.curseforge.com/wow/addons/dejaviewclassic
262939 https://www.curseforge.com/wow/addons/joke
420725 https://www.curseforge.com/wow/addons/keyboards-minimap-icons
33976 https://www.curseforge.com/wow/addons/killtrack
100547 https://www.curseforge.com/wow/addons/plater-nameplates
2849 https://www.curseforge.com/wow/addons/simple-raid-target-icons
413888 https://www.curseforge.com/wow/addons/stay-sheathed-lite
```

**Reply by torkus** [reply]
I wonder if they're safe to install into classic-vanilla wow?

**Reply by layday** [reply]
Some but not others. TBC exposes some C API functions which can't be found in Classic, see https://github.com/Stanzilla/WoWUIBugs/wiki/2.5.1-Consolidated-UI-Changes.

**Reply by torkus** [reply]
looks like addon authors will now need to target 3 versions of WoW :) 

I thought classic-vanilla and classic-bc might share the same engine or something so the addons would work consistently between the two but that seems very far from likely now.

**Reply by torkus** [reply]
marginally interesting:
https://wowpedia.fandom.com/wiki/Public_client_builds#Classics

**Reply by layday** [reply]
I was really hoping to avoid adding another flavour/track but no such luck.

**Reply by torkus** [reply]
Patched version of strongbox released, thanks for the help.

**Reply by layday** [reply]
I don't know if strongbox makes use of `gameVersionFlavour` but there's now a value for TBC: `wow_burning_crusade`.

**Reply by torkus** [reply]
Ah, thanks for the heads-up, it does.

**Reply by layday** [reply]
Tukui has also added a TBC endpoint now, see https://www.tukui.org/api.php.

**Reply by torkus** [reply]
I wonder if wowi will get it's act together. Just saw this: https://www.wowinterface.com/forums/showthread.php?t=58648 , taking a closer read now

**Reply by layday** [reply]
Wouldn't really expect anything to change with WoWI. There's also a new contender in https://addons.wago.io/.

**Reply by torkus** [reply]
Yeah, I was just poking around the wowi API again, looking at the v4 endpoints and their model hasn't changed.

I've seen wago.io a few times now, wondering what their deal is. Looking at the announcement it seems to center around ad revenue sharing. If they put some pressure on Curseforge to compete that's good. If they offer an API for us then that's even better, but I can't shake the feeling these sorts of organisations will lock API access down if they can't pinch some pennies from us.

**Reply by torkus** [reply]
I updated [The List](https://ogri-la.github.io/wow-addon-managers/) last week with a "classic TBC" column for those managers that support it. Let me know when/if you support it.

**Reply by layday** [reply]
Thanks, will do.  Ajour supports classic TBC in the sense that they've added `_classic_beta_` to the list of folders it's able to import. This currently maps to classic - Ajour won't install TBC add-ons from CF or Tukui.  Not sure what the situation is with WowUp.

**Reply by layday** [reply]
WowUp seems to properly support the CF and Tukui classic TBC routes.

**Reply by torkus** [reply]
cheers, @layday , that's helpful. I'll take Ajour off the list.

**Comment by layday** [comment]
@torkus FYI, a new version of [packager](https://github.com/BigWigsMods/packager) attaches a `release.json` file to GitHub releases to distinguish between add-on zips for different game versions - see https://github.com/nebularg/PackagerTest/releases for an example.

**Reply by torkus** [reply]
It might be more useful if it came with a spec. Looking at the commit history it's still all over the place. I'll give it some time to settle down and get established.

just for reference:

```json
{
  "releases": [
    {
      "filename": "TestGit-v1.9.6-classic.zip",
      "nolib": false,
      "metadata": [
        {
          "flavor": "classic",
          "interface": 11307
        }
      ]
    },
    {
      "filename": "TestGit-v1.9.6-nolib.zip",
      "nolib": true,
      "metadata": [
        {
          "flavor": "mainline",
          "interface": 90100
        }
      ]
    },
    {
      "filename": "TestGit-v1.9.6.zip",
      "nolib": false,
      "metadata": [
        {
          "flavor": "mainline",
          "interface": 90100
        }
      ]
    }
  ]
}
```

**Comment by torkus** [comment]
have you heard about wow client support for ~multiple~ per-flavour `.toc` files coming soon? 

1. https://github.com/WeakAuras/WeakAuras2/issues/3118
2. https://github.com/Stanzilla/WoWUIBugs/issues/68#issuecomment-830351390-permalink

**Reply by layday** [reply]
I have, thank you. instawow only deals with TOCs during reconciliation so it doesn't affect me much.
